### ISSUE TITLE
Closing HTTP Client Results in Exception

### ISSUE THREAD

**Issue by masterwok** [DataDog/dd-sdk-flutter#355]
**Describe what happened**

When invoking `close()` on an HTTP client the following exception is thrown:

```
#0      _AssertionError._doThrowNew (dart:core-patch/errors_patch.dart:51:61)
#1      _AssertionError._throwNew (dart:core-patch/errors_patch.dart:40:5)
#2      _DatadogTrackingHttpResponse.listen.<anonymous closure> (package:datadog_tracking_http_client/src/tracking_http_client.dart:482:18)
#3      _HttpClientResponse.listen.<anonymous closure> (dart:_http/http_impl.dart:707:16)
#4      _RootZone.runBinaryGuarded (dart:async/zone.dart:1605:10)
#5      _BufferingStreamSubscription._sendError.sendError (dart:async/stream_impl.dart:358:15)
#6      _BufferingStreamSubscription._sendError (dart:async/stream_impl.dart:376:7)
#7      _BufferingStreamSubscription._addError (dart:async/stream_impl.dart:280:7)
#8      _ForwardingStreamSubscription._addError (dart:async/stream_pipe.dart:128:11)
#9      _addErrorWithReplacement (dart:async/stream_pipe.dart:176:8)
#10     _HandleErrorStream._handleError (dart:async/stream_pipe.dart:274:11)
#11     _ForwardingStreamSubscription._handleError (dart:async/stream_pipe.dart:157:13)
#12     _RootZone.runBinaryGuarded (dart:async/zone.dart:1605:10)
#13     _BufferingStreamSubscription._sendError.sendError (dart:async/stream_impl.dart:358:15)
#14     _BufferingStreamSubscription._sendError (dart:async/stream_impl.dart:376:7)
#15     _BufferingStreamSubscription._addError (dart:async/stream_impl.dart:280:7)
#16     _SyncStreamControllerDispatch._sendError (dart:async/stream_controller.dart:778:19)
#17     _StreamController._addError (dart:async/stream_controller.dart:656:7)
#18     _StreamController.addError (dart:async/stream_controller.dart:610:5)
#19     _HttpParser._reportBodyError (dart:_http/http_parser.dart:1176:22)
#20     _HttpParser._onDone (dart:_http/http_parser.dart:885:9)
#21     _RootZone.runGuarded (dart:async/zone.dart:1581:10)
#22     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:392:13)
#23     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:402:7)
#24     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:291:7)
#25     _SyncStreamControllerDispatch._sendDone (dart:async/stream_controller.dart:782:19)
#26     _StreamController._closeUnchecked (dart:async/stream_controller.dart:637:7)
#27     _StreamController.close (dart:async/stream_controller.dart:630:5)
#28     _Socket._onDone (dart:io-patch/socket_patch.dart:2370:19)
#29     _RootZone.runGuarded (dart:async/zone.dart:1581:10)
#30     _BufferingStreamSubscription._sendDone.sendDone (dart:async/stream_impl.dart:392:13)
#31     _BufferingStreamSubscription._sendDone (dart:async/stream_impl.dart:402:7)
#32     _BufferingStreamSubscription._close (dart:async/stream_impl.dart:291:7)
#33     _SyncStreamControllerDispatch._sendDone (dart:async/stream_controller.dart:782:19)
#34     _StreamController._closeUnchecked (dart:async/stream_controller.dart:637:7)
#35     _StreamController.close (dart:async/stream_controller.dart:630:5)
#36     _RawSecureSocket._close (dart:io/secure_socket.dart:725:17)
#37     _RawSecureSocket.shutdown (dart:io/secure_socket.dart:747:9)
#38     _RawSecureSocket.close (dart:io/secure_socket.dart:703:5)
#39     _Socket._closeRawSocket (dart:io-patch/socket_patch.dart:2327:9)
#40     _Socket.destroy (dart:io-patch/socket_patch.dart:2253:5)
#41     _HttpClientConnection.destroyFromExternal (dart:_http/http_impl.dart:2251:13)
#42     _ConnectionTarget.close (dart:_http/http_impl.dart:2419:11)
#43     _HttpClient._closeConnections (dart:_http/http_impl.dart:2878:24)
#44     _HttpClient.close (dart:_http/http_impl.dart:2639:5)
#45     DatadogTrackingHttpClient.close (package:datadog_tracking_http_client/src/tracking_http_client.dart:206:17)
#46     IOClient.close (package:http/src/io_client.dart:101:15)
#47     EventSourceRepositoryImpl.disconnect (package:CoinMe/data/repositories/event_source_repository_impl.dart:101:28)
#48     Logout.invoke (package:CoinMe/common/usecases/logout.dart:17:34)
<asynchronous suspension>
#49     CommandUseCase.call (package:CoinMe/common/usecases/contracts/command_use_case.dart:9:29)
<asynchronous suspension>
#50     AccountProvider.logOut (package:CoinMe/features/account/account_provider.dart:135:34)
<asynchronous suspension>
#51     AccountScreenState._onLogoutTapped (package:CoinMe/features/account/account_screen.dart:39:7)
<asynchronous suspension>
#52     AccountScreenState.build.<anonymous closure> (package:CoinMe/features/account/account_screen.dart:172:36)
<asynchronous suspension>

```

**Steps to reproduce the issue:**

Install the `datadog_tracking_http_client` version `1.2.0` and enable HTTP tracking on the `DdSdkConfiguration` by invoking `enableHttpTracking()` on the configuration. Create an HTTP client instance and invoke `close()` on the client after the connection has been established. In my use case, the HTTP client being closed is the client being used in package `featurehub_sse_client` and is closed on logout to stop receiving server-sent-events.

**Describe what you expected:**

I would expect an exception not to be thrown. 

**Additional context**
 - Dart/Flutter version: Dart 2.19.2/Flutter 3.7.3
 - Android/iOS OS version: Android API 31
 - Device Model: Android Emulator
 - Datadog SDK version: `datadog_flutter_plugin` version `1.2.2` and `datadog_tracking_http_client` version `1.2.0`
 - Versions of any other relevant dependencies: n/a

**Comment by fuzzybinary** [comment]
Thanks for the report.

I'm taking a look and will try to get a fix out soon.

**Comment by fuzzybinary** [comment]
This is now in `develop` and I'm looking to have a 1.2.1 hotfix with the change out either later today or tomorrow.

Thanks for your patience!

**Comment by fuzzybinary** [comment]
Now available in `datadog_tracking_http_client` 1.2.1. Let me know if you see any issues.
